bundles/org.openhab.binding.smsmodem/src/main/java/org/openhab/binding/smsmodem/internal/handler/SMSModemBridgeHandler.java

```diff
@@ -0,0 +1,473 @@
+        super(bridge);
+        this.serialPortManager = serialPortManager;
+    }
+
+    @Override
+    protected void updateConfiguration(Configuration configuration) {
+        super.updateConfiguration(configuration);
+        scheduler.execute(() -> {
+            modem.stop();
+            checkAndStartModemIfNeeded();
+        });
+    }
+
+    @Override
+    public void initialize() {
+        shouldRun = true;
+        if (checkScheduled == null || (checkScheduled.isDone()) && this.shouldRun) {
+            checkScheduled = scheduler.scheduleWithFixedDelay(this::checkAndStartModemIfNeeded, 0, 15,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    private void checkAndStartModemIfNeeded() {
+        try {
+            if (shouldRun && !isRunning()) {
+                logger.debug("Initializing smsmodem");
+                // ensure the underlying modem is stopped before trying to (re)starting it :
+                if (modem != null) {
+                    modem.stop();
+                }
+                String logName;
+                if (getThing().getThingTypeUID().equals(SMSModemBindingConstants.SMSMODEMBRIDGE_THING_TYPE)) {
+                    SMSModemBridgeConfiguration config = getConfigAs(SMSModemBridgeConfiguration.class);
+                    modem = new Modem(serialPortManager, resolveEventualSymbolicLink(config.serialPort),
+                            Integer.valueOf(config.baud), config.simPin, scheduler, config.pollingInterval,
+                            config.delayBetweenSend);
+                    checkParam(config);
+                    logName = config.serialPort + " | " + config.baud;
+                } else if (getThing().getThingTypeUID()
+                        .equals(SMSModemBindingConstants.SMSMODEMREMOTEBRIDGE_THING_TYPE)) {
+                    SMSModemRemoteBridgeConfiguration config = getConfigAs(SMSModemRemoteBridgeConfiguration.class);
+                    modem = new Modem(serialPortManager, resolveEventualSymbolicLink(config.ip),
+                            Integer.valueOf(config.networkPort), config.simPin, scheduler, config.pollingInterval,
+                            config.delayBetweenSend);
+                    checkRemoteParam(config);
+                    logName = config.ip + ":" + config.networkPort;
+                } else {
+                    throw new IllegalArgumentException("Invalid thing type");
+                }
+                logger.debug("Now trying to start SMSModem {}", logName);
+                modem.registerStatusListener(this);
+                modem.registerMessageListener(this);
+                modem.registerInformationListener(this);
+                modem.start();
+                logger.debug("SMSModem {} started", logName);
+            }
+        } catch (ModemConfigurationException e) {
+            String message = e.getMessage();
+            logger.error(message, e);
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.

Reply:
Done